src/main/java/com/gtnewhorizons/angelica/glsm/debug/OpenGLDebugging.java

```diff
@@ -0,0 +1,441 @@
+package com.gtnewhorizons.angelica.glsm.debug;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.FloatBuffer;
+import java.nio.IntBuffer;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Consumer;
+
+import com.gtnewhorizons.angelica.glsm.GLStateManager;
+import org.apache.commons.io.FileUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.lwjgl.BufferUtils;
+import org.lwjgl.opengl.GL11;
+
+// Adapted from https://github.com/makamys/CoreTweaks/blob/master/src/main/java/makamys/coretweaks/util/OpenGLDebugging.java;
+// originally from https://github.com/TheGreyGhost/MinecraftByExample/blob/1-8final/src/main/java/minecraftbyexample/usefultools/OpenGLdebugging.java
+
+public class OpenGLDebugging {
+
+    public static final Logger LOGGER = LogManager.getLogger("gldumper");
+
+    private static final Consumer<String> LINE_LOGGER = LOGGER::debug;
+    public static final String GL_IS_ENABLED = "glIsEnabled()";
+    public static final String GL_GET_BOOLEANV = "glGetBooleanv()";
+    public static final String GL_GET_INTEGERV = "glGetIntegerv()";
+    public static final String GL_GET_FLOATV = "glGetFloatv()";
+
+    public class GLproperty {
+
+        public GLproperty(int init_gLconstant, String init_name, String init_description, String init_category, String init_fetchCommand) {
+            gLconstant = init_gLconstant;
+            name = init_name;
+            description = init_description;
+            category = init_category;
+            fetchCommand = init_fetchCommand;
+        }
+
+        public int gLconstant;
```

Comment:
I guess this is subjective but I find this capitalization pretty odd, `glConstant` would look nicer and be consistent with how the gl methods are capitalized.

This also applies to the occurrences of `gLproperty` later in the file.